src/Display/RGBColor.hpp

```diff
@@ -51,6 +51,53 @@ namespace Display
 		{
 			return 0.3f * (color >> 16) + 0.59f * (color >> 8 & 0xFF) + 0.11f * (color & 0xFF);
 		}
+
+		constexpr RGBColor operator+(RGBColor other)
+		{
+			return RGBColor(other.r + r, other.g + g, other.b + b);
+		}
+
+		constexpr RGBColor operator-(RGBColor other)
+		{
+			return RGBColor(other.r - r, other.g - g, other.b - b);
+		}
+
+		constexpr RGBColor operator*(RGBColor other)
```

Comment:
This is a code smell, it shouldn't be multiplying them directly, take a look at the `*=` operator for what it should actually be doing after I fix it :) 

I wouldn't call it mixing, it's in essence just that, multiplication

I guess you could call it applying a bias? it's basically per component multiplication, this is much easier and more intuitive if you represent them in float form,

`RGBColor(0.5f, 0.5f, 0.5f) * RGBColor(1f, 0f, 0f)` is `obviously RGBColor(0.0f, 0f, 0f)`, the byte representation is the exact same but each component goes from 0 - 255, remember thats just how you represent it, you're not actually doing 255*255 (because that'd obviously overflow, which is what will happen in this code smell)

Reply:
Ah, I see. That looks like the "multiply blend", so perhaps all the overloaded operators can be replaced with:

```cpp
static constexpr RGBColor RGBColor::BlendMultiply(RGBColor a, RGBColor b) { return a.Mix(b); }
constexpr RGBColor& RGBColor::BlendMultiply(RGBColor b) {...}
```

I think it's more expressive, and should result in the same codegen.